src/ctfeexpr.c

```diff
@@ -813,10 +813,12 @@ UnionExp pointerArithmetic(Loc loc, TOK op, Type *type,
         goto Lcant;
     }
 
+    // Create a CTFE pointer &agg1[indx]
     IntegerExp *ofs = new IntegerExp(loc, indx, Type::tsize_t);
-    new(&ue) IndexExp(loc, agg1, ofs);
-    IndexExp *ie = (IndexExp *)ue.exp();
-    ie->type = type;
+    Expression *ie = new IndexExp(loc, agg1, ofs);
+    ie->type = pointee;
+    new(&ue) AddrExp(loc, ie);
```

Comment:
It did when Walter first started using placement new, but I've added support for it to the converter.